compiler/hash-types/src/new/data.rs

```diff
@@ -81,6 +82,53 @@ pub struct CtorPat {
     pub args: DefPatArgsId,
 }
 
+/// A numeric constructor definition.
+///
+/// This is a constructor which accepts numeric literals
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct NumericCtorInfo {
+    /// The number of bits in the number.
+    pub bits: u16,
```

Comment:
A little large for number of bits, maybe a `u8`?

Reply:
Fine, probably won't make any difference due to aligning

Reply:
If not, we can just keep it like this